File: trail1/규칙을 찾아 출력하기/find-a-rule-and-print.c
#include <stdio.h>

int main() {
    int N;
    scanf("%d", &N);
    char board[N][N];
    for (int i=0; i<N; i++) {
        for (int j=0; j<N; j++) {
            if (i==0 || i==N-1 || j==0 || j==N-1) {board[i][j] = '*';}
            else if (i > j) {board[i][j] = '*';}
            else {board[i][j] = ' ';}
        }
    }
    for (int i=0; i<N; i++) {
        for (int j=0; j<N; j++) {
            printf("%c ", board[i][j]);
        }
        printf("\n");
    } 
    return 0;
}

File: trail1/규칙을 찾아 출력하기/find-a-rule-and-print.py
N = int(input())
board = [[' ' for _ in range(N)] for _ in range(N)]
for i in range(N):
    for j in range(N):
        if i==0 or i==N-1 or j==0 or j==N-1:
            board[i][j] = '*'
        elif i > j:
            board[i][j] = '*'
        else:
            board[i][j] = ' '

for i in range(N):
    for j in range(N):
        print(board[i][j], end=" ")
    print()